lib/pf/util/dns.pm

```diff
@@ -33,31 +33,55 @@ Returns 2 values
 
 sub matches_passthrough {
     my ($domain, $zone) = @_;
+
+    return _matches_passthrough(eval("\\%$zone"), $domain);
```

Comment:
Avoid the use of eval as it is unsafe.

A safer approach is to create a lookup table for which hash to use.

Reply:
Example

```bash
perl -I/usr/local/pf/lib -Mpf::util::dns -e'pf::util::dns::matches_passthrough(undef, "passthroughs,print(\"gotcha\\n\"),\%passthroughs");'```